Actions/Deliver/Deliver.ps1

```diff
@@ -146,359 +157,346 @@ try {
-                    $projectSettings.deliverToAppSource."$_" = $projectSettings."AppSource$_"
-                }
+            if ($artFolder.Count -gt 1) {
+                $artFolder | Out-Host
+                throw "Internal error - multiple apps folders located"
             }
-            # if type is Release, we only get here with the projects that needs to be delivered to AppSource
-            # if type is CD, we get here for all projects, but should only deliver to AppSource if AppSourceContinuousDelivery is set to true
-            if ($type -eq 'Release' -or $projectSettings.deliverToAppSource.continuousDelivery) {
-                EnsureAzStorageModule
-                $appSourceContext = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($secrets.appSourceContext)) | ConvertFrom-Json | ConvertTo-HashTable
-                if (!$appSourceContext) {
-                    throw "appSourceContext secret is missing"
-                }
-                $authContext = New-BcAuthContext @appSourceContext
-
-                if ($projectSettings.deliverToAppSource.MainAppFolder) {
-                    $AppSourceMainAppFolder = $projectSettings.deliverToAppSource.MainAppFolder
-                }
-                else {
-                    try {
-                        $AppSourceMainAppFolder = $projectSettings.appFolders[0]
-                    }
-                    catch {
-                        throw "Unable to determine main App folder"
-                    }
-                }
-                if (!$projectSettings.deliverToAppSource.ProductId) {
-                    throw "deliverToAppSource.ProductId needs to be specified in $thisProject/.AL-Go/settings.json in order to deliver to AppSource"
-                }
-                Write-Host "AppSource MainAppFolder $AppSourceMainAppFolder"
+            $artfolder = $artfolder[0].FullName
+            $appFile = Get-ChildItem -path $artFolder | Where-Object { $_.name -like $mainAppFileName } | ForEach-Object { $_.FullName }
+            $libraryAppFiles = @(Get-ChildItem -path $artFolder | Where-Object { $_.name -notlike $mainAppFileName } | ForEach-Object { $_.FullName })
 
-                $mainAppJson = Get-Content -Path (Join-Path $baseFolder "$thisProject/$AppSourceMainAppFolder/app.json") -Encoding UTF8 | ConvertFrom-Json
-                $mainAppFileName = ("$($mainAppJson.Publisher)_$($mainAppJson.Name)_".Split([System.IO.Path]::GetInvalidFileNameChars()) -join '') + "*.*.*.*.app"
-                $artfolder = @(Get-ChildItem -Path (Join-Path $artifactsFolder "$project-$refname-Apps-*.*.*.*") | Where-Object { $_.PSIsContainer })
-                if ($artFolder.Count -eq 0) {
-                    throw "Internal error - unable to locate apps folder"
+            $appSourceIncludeDependencies = $projectSettings.deliverToAppSource.includeDependencies
```

Comment:
Where do these changes come from?

Reply:
Good catch :) Those came from a bad merge. Should be fixed now